brainscore_vision/model_helpers/brain_transformation/__init__.py

```diff
@@ -30,12 +36,18 @@ def __init__(self, identifier,
         self.activations_model._extractor.set_visual_degrees(visual_degrees)  # for microsaccades
         self._visual_degrees = visual_degrees
         # region-layer mapping
+
+        # Attempt to load region_layer_map from JSON, if available
+        region_layer_map = self.load_region_layer_map_json(identifier) if region_layer_map is None else region_layer_map
```

Comment:
I don't like that the `ModelCommitment` has to know about plugins. Can we not load the specification inside the plugin and pass it to the `ModelCommitment` as a parameter?

Reply:
Modifying the `__init__.py` was the original approach. However, after talking with the others, it seems modifying the init file, especially when there can be multiple model identifiers listed in a single model directory could be messy. It would also make for a messier PR/commit during the model submission.

By adding a check in `ModelCommitment`, all that would be needed would be a single region_layer_map json per model_identifier in the model directory. But I do agree that it disrupts the current flow of the plugin model.

If we do not want to modify the `__init__.py` file of each model (and therefore cannot pass the map to `ModelCommitment` as a parameter), what would you recommend?

Reply:
just thinking out loud, but what about an intermediate approach where the `ModelCommitment` expects the full layer mapping, but submissions can specify this to be automatically discovered? Something like
```
model_registry['mymodel'] = lambda: ModelCommitment(
    identifier='mymodel',
    activations_model=get_model(),
    region_layer_map=discover_region_layer_mapping('mymodel'))
```

The `discover_region_layer_mapping` would then first look for a mapping file in the plugin, and if that doesn't exist make one by running the layer selection. This way the plugin `__init__` would not have to be modified but we still maintain the separation between plugins and model helpers.

Reply:
Martin - Kartik and I just discussed and I would like to propose something a bit more direct that what you suggested.  We create a means to determine if we are running locally (any place but AWS) or in production (AWS). For local runs (again, anything but AWS) if no json file exists, we perform layer mapping and then write the json file. 
 For AWS runs we assume that region_layer_map json file exists and throw an exception if it does not, terminating the scoring run.

This is changing the assumptions of modelCommitment, which hopefully we don't do frequently, but not significantly (and not even as significantly as the solution you propose above).  Additionally, as a benefit, local users, at the time of their PR already have the layer map and have thought about it and checked it against reality.  They can regenerate as many times as desired, simply by removing the json file.

Reply:
yes that sounds good to me. And keeps the decoupling to a decent extent.